presto-docs/src/main/sphinx/connector/kafka.rst

```diff
@@ -259,6 +264,430 @@ Field           Required  Type      Description
+Presto data type                      ``dataFormat`` values
+===================================== =======================================
+``BIGINT``                            ``BYTE``, ``SHORT``, ``INT``, ``LONG``
+``INTEGER``                           ``BYTE``, ``SHORT``, ``INT``
+``SMALLINT``                          ``BYTE``, ``SHORT``
+``TINYINT``                           ``BYTE``
+``DOUBLE``                            ``DOUBLE``, ``FLOAT``
+``BOOLEAN``                           ``BYTE``, ``SHORT``, ``INT``, ``LONG``
+``VARCHAR`` / ``VARCHAR(x)``          ``BYTE``
+===================================== =======================================
+
+The ``mapping`` attribute specifies the range of bytes in a key or
+message used for encoding. It can be one or two numbers separated by a colon
+(``<start>[:<end>]``).
+
+.. note::
+
+    For ``VARCHAR`` types a start and end position must be defined. Otherwise
+    there is no way to know how many bytes the message contains. The raw format
+    mapping information is static and cannot be dynamically changed to fit the
+    variable width of some Presto data types.
+
+If only a start position is given:
+
+* For fixed width types the appropriate number of bytes are used for the
+  specified ``dateFormat`` (see above).
+
+If start and end position are given, then:
+
+* For fixed width types the size must be equal to number of bytes used by
+  specified ``dataFormat``.
+* For ``VARCHAR`` all bytes between start (inclusive) and end (exclusive) are
+  used.
+
+.. note::
+
+    At the very least a start value must be present in the mapping for encoding
+    to work.
+
+Encoding scheme of numeric data types (``BIGINT``, ``INTEGER``, ``SMALLINT``,
+``TINYINT``, ``DOUBLE``) is straightforward. Table column values are encoded to
+bytes according to either:
+
+* big-endian encoding (for integer types)
+* IEEE 754 format for (for ``DOUBLE``).
+
+Example raw field definition in a :ref:`table definition file<kafka-table-definition-files>`
+for a Kafka message:
+
+.. code-block:: none
+
+    "message": {
+            "dataFormat": "raw",
+            "fields": [
+                {
+                    "name": "field1",
+                    "type": "BIGINT",
+                    "dataFormat": "LONG",
+                    "mapping": "0"
+                },
+                {
+                    "name": "field2",
+                    "type": "INTEGER",
+                    "dataFormat": "INT"
+                    "mapping": "8"
+                },
+                {
+                    "name": "field3",
+                    "type": "SMALLINT",
+                    "dataFormat": "LONG",
+                    "mapping": "12"
+                },
+                {
+                    "name": "field4",
+                    "type": "VARCHAR(6)",
+                    "dataFormat": "BYTE",
+                    "mapping": "20:26"
+                }
+            ]
+    }
+
+.. note::
```

Comment:
Let's make this a non-note, since this seems like normal explanatory text. Having too many notes makes documentation harder to read (and reduces the values of a note).